diff --git a/public/question/type/numerical/questiontype.php b/public/question/type/numerical/questiontype.php
index 387073d135b..53323a7a897 100644
--- a/public/question/type/numerical/questiontype.php
+++ b/public/question/type/numerical/questiontype.php
@@ -229,7 +229,7 @@ class qtype_numerical extends question_type {
             }
             $options->question = $question->id;
             $options->answer   = $answer->id;
-            if (trim($question->tolerance[$key]) == '') {
+            if (!array_key_exists($key, $question->tolerance) || trim($question->tolerance[$key]) == '') {
                 $options->tolerance = '';
             } else {
                 $options->tolerance = $this->apply_unit($question->tolerance[$key],
